tracer/src/Datadog.Trace/Configuration/IntegrationSettings.cs

```diff
@@ -44,21 +44,21 @@ public IntegrationSettings(string integrationName, IConfigurationSource source)
         public string IntegrationName { get; }
 
         /// <summary>
-        /// Gets or sets a value indicating whether
+        /// Gets a value indicating whether
         /// this integration is enabled.
         /// </summary>
-        public bool? Enabled { get; set; }
+        public bool? Enabled { get; internal set; }
```

Comment:
I don't think we should remove this setters. This is how users can disable integrations via code (see [docs](https://docs.datadoghq.com/tracing/setup_overview/setup/dotnet-framework/?tab=code#automatic-instrumentation-integration-configuration)). Same for configuring analytics below, although those we could mark those obsolete.

```csharp
var settings = TracerSettings();
settings.Integrations[name].Enabled = false;

Tracer.Instance = new Tracer(settings);
```

Reply:
That's not the "public" API to disable integrations though, right? There's the public `DisabledIntegrationNames` setting for that:

```csharp
var settings = TracerSettings();
settings.DisabledIntegrationNames.Add(name);

Tracer.Instance = new Tracer(settings);
```

It's confusing to have both IMO, though "immutable" settings may solve some of it.

Reply:
I think immutable settings resolves my concerns around these anyway actually, so I reverted the changes to `IntegrationSettings` in this PR. 

> There's the public DisabledIntegrationNames setting .. It's confusing to have both IMO

I stand by this though 😉 Anything we can do there? should we make `DisabledIntegrationNames` obsolete?

Reply:
`DD_TRACE_<integration>_ENABLED` (and the `TracerSettings.Integrations[name].Enabled` property in our case) is the standard setting across all language tracers. `DisabledIntegrationNames` predates the "official" setting and should be deprecated.

There's another wrinkle here: what does it mean to "disable" an integration? With "call site," the native code only supported the _environment variable_ `DD_DISABLED_INTEGRATIONS`. When disabled this way, we could skip rewriting IL entirely, which was safer and faster. For example, if an the IL modifications from one integration was cause apps to crash, users could disable the integration this way and avoid the crash. Configuration via code or file, or using `DD_TRACE_<integration>_ENABLED` (even the env var), would only apply in managed code, _after_ IL rewriting already happened. We still modify the IL and call into the integration code, we just don't generate a scope/span.

[I had a branch](https://github.com/DataDog/dd-trace-dotnet/commit/56df421ca8ed5febe1eca27f600fc91e5c4b9092) were I was adding support for `DD_TRACE_<integration>_ENABLED` in native code, but at this point, since call-target make ReJIT requests from managed code, we would have to skip those ReJIT requests instead to avoid rewriting IL.

Reply:
Yeah, that makes sense. I actually thought we already skipped the rewriting if `DD_TRACE_<integration>_ENABLED` was false, but I don't know if we should, otherwise we can have a position where they re-enable the integration in code, but it won't necessarily work correctly if we've already missed our chance to rewrite. Anyway, I think that's a separate feature we could look into.